dev/update_changelog.py

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+# -*- coding: utf-8 -*-
+"""Script for automatically updating the `CHANGELOG.md` based on the commits since the latest release tag."""
+from pathlib import Path
+import re
+import subprocess
+
+DEFAULT_CHANGELOG_SECTIONS = """
+### ‼️ Breaking changes
+
+
+### ✨ New features
+
+
+### 🗑️ Deprecation
+
+
+### 👌 Improvements
+
+
+### 🐛 Bug Fixes
+
+
+### 📚 Documentation
+
+
+### 🔧 Maintenance
+
+
+### ⬆️ Update dependencies
+
+
+### ♻️ Refactor
+
+"""
+
+
+def update_changelog():
+    """Update the `CHANGELOG.md` for a first draft of the release."""
+
+    print('🔍 Checking the current version number')
+    with Path('CHANGELOG.md').open('r', encoding='utf8') as handle:
+        current_changelog = handle.read()
+
+    with Path('src/aiida_quantumespresso/__init__.py').open('r', encoding='utf8') as handle:
+        version = re.search(r"__version__ = '(?P<version_number>\d+.+)'", handle.read()).groupdict()['version_number']
+
+    if str(version) in current_changelog:
+        print('🛑 Current version already in `CHANGELOG.md`. Skipping...')
+        return
+
+    print('⬆️ Found updated version number, adapting `CHANGELOG.md`.')
+    tags = subprocess.run(['git', 'tag'], capture_output=True, check=False).stdout
+
+    tag_pattern = re.compile(r'(v\d\.\d\.\d)\n')
+    tags = tags.decode()
+
+    latest_tag = tag_pattern.findall(tags)[-1]
+
+    print(f'🔄 Comparing with latest tag `{latest_tag}`.')
+    commits = subprocess.run(['git', 'log', "--pretty=format:'%s %h %H'", f'{latest_tag}..origin/main'],
```

Comment:
How about you use `--pretty-format:'%h|%H|%s'`. You can then later use:
```
hash_short, hash_long, message = commit.split('|', maxsplit=2)
```
Think that is a bit cleaner and faster as you don't split and re-join the message.